irr/src/COpenGLDriver.cpp

```diff
@@ -3018,7 +3018,8 @@ IImage *COpenGLDriver::createScreenShot(video::ECOLOR_FORMAT format, video::E_RE
 	if (newImage)
 		pixels = static_cast<u8 *>(newImage->getData());
 	if (pixels) {
-		glReadBuffer(GL_FRONT);
+		if (!Params.Doublebuffer)
```

Comment:
Suggestion: Write this as `glReadBuffer(Params.Doublebuffer ? GL_BACK : GL_FRONT);` instead. Then you can get rid of the somewhat dubious `glReadBuffer(GL_BACK);` a few lines below.

Reply:
I was under the impression that no, but after a closer look, there are two calls to `glCopyTexImage2D` which *might* rely on this. So it's probably safer to restore it. Still, two thoughts:

* "It is an error to specify a nonexistent buffer to glReadBuffer" - shouldn't we restore to `GL_FRONT` if we are to support a singlebuffer configuration?
* I wonder whether this fix should also be applied to the OGLES driver.

Reply:
> * "It is an error to specify a nonexistent buffer to glReadBuffer" - shouldn't we restore to `GL_FRONT` if we are to support a singlebuffer configuration?

Good point. I applied your suggestion.



> * I wonder whether this fix should also be applied to the OGLES driver.

I just tested it and I can't reproduce the bug, but maybe it's just luck and it will resurface in a different hardware/sofrware configuration. Should I apply an equivalent patch anyway?

Reply:
> I just tested it and I can't reproduce the bug, but maybe it's just luck and it will resurface in a different hardware/sofrware configuration. Should I apply an equivalent patch anyway?

I'd leave it untouched in that case.